File: assignment3/python/beanbag.py
from furniture import Furniture

class BeanBag(Furniture): 
    DEFAULT_MATERIAL : str = 'cloth'
    DEFAULT_DIAMETER : float = 48.0 #inches
    def __init__(self, material : str = DEFAULT_MATERIAL, diameter : float = DEFAULT_DIAMETER):
        super(BeanBag, self).__init__(material)
        self._diameter = diameter
        
    @property
    def size(self):
        return self._diameter

File: assignment3/c++/src/beanbag.cpp
#include "beanbag.h"

static const double DEFAULT_DIAMETER = 48.0;

BeanBag::BeanBag(std::string material, double diameter = DEFAULT_DIAMETER) : Furniture(material) {
    BeanBag::_material = material;
    BeanBag::_diameter = diameter;
}

double BeanBag::getDiameter() const { return _diameter; }